Radio.m

```diff
@@ -1445,14 +1577,62 @@ - (void) parseGpsToken: (NSScanner *) scan {
     
 }
 
-
+/**
+ * Parse Profile tokens
+ *     called on the GCD thread associated with the GCD tcpSocketQueue
+ *
+ *     format: <apiHandle>|profile <profileType> list=<value>^<value>^...<value>^
+ *                           OR
+ *     format: <apiHandle>|profile <profileType> current=<value>
+ *
+ *     scan is initially at scanLocation = 17, start of the <profileType>
+ *     "<apiHandle>|profile " has already been processed
+ */
 - (void) parseProfileToken: (NSScanner *) scan {
-    
+    // get the Profile type
+    NSString *profileType;
+    [scan scanUpToString:@" " intoString: &profileType];
+    // skip the space
+    [scan scanString:@" " intoString: nil];
+    
+    // get the Sub type
+    NSString *profileSubType;
+    [scan scanUpToString:@"=" intoString: &profileSubType];
+    // skip the "="
+    [scan scanString:@"=" intoString: nil];
+    
+    // get the remainder of the command
+    NSString *remainderOfCommand;
+    [scan scanUpToString:@"\n" intoString: &remainderOfCommand];
+    
+    // List or Current value?
+    if ([profileSubType isEqualToString: @"list"]) {
+        // it's the List, separate the components
+        NSMutableArray *profileNames;
```

Comment:
Good for these to be mutable as they will have profiles added/removed dynamically